client/features/routineSlice.js

```diff
@@ -38,76 +38,81 @@ export const createRoutine = createAsyncThunk(
   }
 );
 
-export const editRoutine = createAsyncThunk(
-  "routine/editRoutine",
+export const updateRoutine = createAsyncThunk(
+  'routine/editRoutine',
   async ({ userId, name, duration }) => {
-    const res = await axios.post(`/routine`, { userId, name, duration });
+    const res = await axios.put(`/routine`, { userId, name, duration });
```

Comment:
this needs an update to the route. Check the backend for what it needs. Also argument names should be edited.